src/constraints/GeomMeanEpiCone.jl

```diff
@@ -5,146 +5,83 @@
-    )
-        return GeomMeanEpiCone(A, B, t // 1, fullhyp)
+        return new(t, side_dimension)
     end
 end
 
-mutable struct GeomMeanEpiConeConstraint <: Constraint
-    T::AbstractExpr
-    cone::GeomMeanEpiCone
+MOI.dimension(set::GeomMeanEpiConeSquare) = 3 * set.side_dimension
 
-    function GeomMeanEpiConeConstraint(T::AbstractExpr, cone::GeomMeanEpiCone)
-        if size(T) != cone.size
-            throw(DimensionMismatch("T must be size $(cone.size)"))
-        end
-        return new(T, cone)
-    end
+head(io::IO, ::GeomMeanEpiConeSquare) = print(io, "∈(GeomMeanEpiConeSquare)")
 
-    function GeomMeanEpiConeConstraint(T::Value, cone::GeomMeanEpiCone)
-        return GeomMeanEpiConeConstraint(constant(T), cone)
+function Base.in(func::Tuple, set::GeomMeanEpiConeSquare)
+    for f in func
+        n = LinearAlgebra.checksquare(f)
+        if n != set.side_dimension
+            throw(
+                DimensionMismatch(
+                    "Matrix of side dimension `$n` does not match set of side dimension `$(set.side_dimension)`",
+                ),
+            )
+        end
     end
-end
-
-head(io::IO, ::GeomMeanEpiConeConstraint) = print(io, "∈(GeomMeanEpiCone)")
-
-Base.in(T, cone::GeomMeanEpiCone) = GeomMeanEpiConeConstraint(T, cone)
-
-function AbstractTrees.children(constraint::GeomMeanEpiConeConstraint)
-    return (
-        constraint.T,
-        constraint.cone.A,
-        constraint.cone.B,
-        "t=$(constraint.cone.t)",
-    )
+    return GenericConstraint(vcat(vec.(func)...), set)
 end
 
 # For t ∈ [-1, 0] ∪ [1, 2] the t-weighted matrix geometric mean is matrix convex
 # (arxiv:1512.03401).
 # So if A and B are convex sets, then T ⪰ A #_t B will be a convex set.
-function vexity(constraint::GeomMeanEpiConeConstraint)
-    A = vexity(constraint.cone.A)
-    B = vexity(constraint.cone.B)
-    T = vexity(constraint.T)
-
-    # NOTE: can't say A == NotDcp() because the NotDcp constructor prints a
-    # warning message.
-    if typeof(A) == ConcaveVexity || typeof(A) == NotDcp
-        return NotDcp()
-    elseif typeof(B) == ConcaveVexity || typeof(B) == NotDcp
-        return NotDcp()
-    end
-    vex = ConvexVexity() + (-T)
-    if vex == ConcaveVexity()
+function vexity(constraint::GenericConstraint{GeomMeanEpiConeSquare})
+    n = constraint.set.side_dimension
+    T = vexity(constraint.child[1:n^2])
+    A = vexity(constraint.child[n^2 .+ (1:n^2)])
+    B = vexity(constraint.child[2n^2 .+ (1:n^2)])
+    if A in (ConcaveVexity(), NotDcp()) || B in (ConcaveVexity(), NotDcp())
         return NotDcp()
     end
-    return vex
+    return ConvexVexity() + (-T)
 end
 
 function _add_constraint!(
     context::Context,
-    constraint::GeomMeanEpiConeConstraint,
+    constraint::GenericConstraint{GeomMeanEpiConeSquare},
 )
-    A = constraint.cone.A
-    B = constraint.cone.B
-    t = constraint.cone.t
-    T = constraint.T
+    n = constraint.set.side_dimension
+    T = reshape(constraint.child[1:n^2], n, n)
```

Comment:
It may seems silly to vectorize to fit into the `GenericConstraint` and then have to `reshape` here but in the long term, this `_add_constraint!` will be replaced by bridges which will work with the vectorized version anyway.

Reply:
reshape should be cheap also, it only updates the `size` metadata, it doesn't touch the data (since everything is also vectorized inside `conic_form!`s)